File: Pascal's Triangle/Pascal's Triangle.py
class Solution:
    # @return a list of lists of integers
    def generate(self, numRows):
        list=[]
        if numRows==0:
            return list
        list.append([1])
        for i in range(1, numRows):
            last=list[len(list)-1]
            value=[]
            for k in range(i+1):
                if k==0 or k==i:
                    value.append(1)
                else:
                    value.append(last[k-1]+last[k])
            list.append(value)
        return list
                    
                    

File: Pascal's Triangle/Pascal's Triangle.java
public class Solution { 
    public List<List> generate(int numRows) {
        List<List> list = new ArrayList<List>();
        if( numRows== 0 ) return list;
        for ( int i = 0; i < numRows; i++ ){
            List<Integer> value = new ArrayList<Integer>();
            if ( i == 0 ){
                value.add(1);
                list.add(value);
                continue;
            } else {
                List<Integer> last = list.get(list.size()-1);
                for ( int k = 0; k <= i; k++ ){
                    if ( k == 0 ) value.add(1);
                    else if( k == i ) value.add(1);
                    else{
                        value.add(last.get(k-1)+last.get(k));
                    }
                } 
                list.add(value);
            }
        }
        return list;
    }
}
